fix: flake8-bugbear code B024

fapont opened the pull request:
See #106571 item B024

This fix concerns the addition of `abstractmethod` to methods declared inside abstract classes.

Should I also include PEP8 compliant reformatting on the files I had to modify ?

fapont committed:
fix: flake8-bugbear code B024

fapont commented:
I'm not sure to understand why my commit is not authorized even after several attempt to generate my CLA

kit1980 commented:
Looks good to me, thanks.
Regarding re-formatting, the formatting should be already correct.
If not for some reason, then it should a separate PR.

kit1980 commented:
@pytorchbot merge -r

kit1980 commented:
I removed "Fixes" from the description because we don't want to close the issue yet (and GitHub will autoclose seeing "Fixes").

kit1980 commented:
@fapont There is an error, please see https://github.com/pytorch/pytorch/actions/runs/6345393257/job/17237391159?pr=107265

> TypeError: Can't instantiate abstract class ConfigurableQuantizeHandler with abstract methods is_custom_module, is_standalone_module

kit1980 commented on torch/ao/quantization/fx/quantize_handler.py:
```diff
@@ -94,9 +95,11 @@ def is_general_tensor_value_op(self) -> bool:
         """
         return False
 
+    @abstractmethod
```
This should not be marked abstract.

kit1980 commented on torch/ao/quantization/fx/quantize_handler.py:
```diff
@@ -94,9 +95,11 @@ def is_general_tensor_value_op(self) -> bool:
         return self.is_custom_module_
 
+    @abstractmethod
     def is_standalone_module(self):
```
This should not be marked abstract.

kit1980 commented on torch/ao/quantization/fx/quantize_handler.py:
```diff
@@ -77,6 +77,7 @@ def __init__(
                             arg, self.modules, cache_for_no_tensor_check)):
                     self.num_tensor_args += 1
 
+    @abstractmethod
```
Doesn't look abstract to me.

fapont committed:
fix: remove useless abstracmethod decorator

fapont commented:
Hi @kit1980, thank you for your feedback. I removed unnecessary decorator. I'm waiting for the tests to run and then try to find something else to do :)

fapont commented:
A problem.. I'll take a look at it and suggest a solution when I get home

kit1980 commented:
> A problem.. I'll take a look at it and suggest a solution when I get home

I'd say just add # noqa: B024 for those. Not every abstract class must have abstract methods.

fapont committed:
fix: ignore B024 on QuantizeHandler class

fapont committed:
fix: fix ruff and flake8 warnings

fapont committed:
fix: mypy linting error

fapont commented:
This time it should be good.
- I fixed few warnings by adding "# noqa: B024"
- I fixed instance creation error by implementing a dummy function here https://github.com/pytorch/pytorch/pull/107265/files#diff-bb87e3b609df0eee1ef22dad1fd3a122cc5b0102106e5451f60cae0ef3dbc07bR75
- I added `isort` suggestion on 1 file (auto formatting on other would change too much things and affect PR review visibility)

kit1980 commented:
@pytorchbot merge
